src/tabs/tabs.js

```diff
@@ -160,21 +184,19 @@ angular.module('ui.bootstrap.tabs', [])
 
       scope.select = function(evt) {
         if (!scope.disabled) {
-          var index;
-          for (var i = 0; i < tabsetCtrl.tabs.length; i++) {
-            if (tabsetCtrl.tabs[i].tab === scope) {
-              index = i;
-              break;
-            }
-          }
-
-          tabsetCtrl.select(index, evt);
+          tabsetCtrl.select(tabsetCtrl.findTabIndex(scope), evt);
         }
       };
 
       tabsetCtrl.addTab(scope);
       scope.$on('$destroy', function() {
         tabsetCtrl.removeTab(scope);
+        if (disableWatcher) {
+          disableWatcher();
```

Comment:
Need to set to `null` after executing to avoid holding references in memory (gc issue)